minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-framework-core/src/main/java/org/apache/nifi/minifi/c2/command/DefaultUpdateConfigurationStrategy.java

```diff
@@ -148,7 +150,7 @@ private void reloadFlow(Set<String> proposedConnectionIds) throws IOException {
         List<ValidationResult> validationErrors = validate(flowController.getFlowManager());
         if (!validationErrors.isEmpty()) {
             LOGGER.error("Validation errors found when reloading the flow: {}", validationErrors);
-            throw new IllegalStateException("Unable to start flow due to validation errors");
+            throw new ValidationException(validationErrors);
```

Comment:
I would keep the error message here as well.
Otherwise when creating the operation response, the exception message will be null. It is more informative for the user if they get a message that the flow did not start due to a validation error.

For the the ValidationException class also needs to be modified:

```
public class ValidationException extends IllegalStateException {
    private List<ValidationResult> validationResults;

    public ValidationException(String message, List<ValidationResult> details) {
        super(message);
        this.validationResults = details;
    }
```

Reply:
Fixed, thanks.